File: src/ex/lib/phantasy_island_remote_control/image_process.ex
defmodule PhantasyIslandRemoteControl.ImageProcess do
  @moduledoc """
  这个文件在 PhantasyIslandRemoteControl 库中负责解析从仿真平台发回的无人机相机图像
  """

  def read_b64_img(nil), do: nil

  def read_b64_img(uri) do
    # "data:image/png;base64,..."
    case String.split(uri, ",") do
      [_, b64_data] ->
        Base.decode64!(b64_data)
        # 在 Elixir 中，通常返回二进制数据。如果需要进一步处理，可能需要 OpenCV 的 NIFs (如 Evision)
      _ ->
        nil
    end
  end
end

File: src/el/image_process.erl
-module(image_process).
-export([read_b64_img/1]).

read_b64_img(nil) -> nil;
read_b64_img(URI) when is_binary(URI) ->
    case binary:split(URI, <<",">>) of
        [_, B64] ->
            base64:decode(B64);
        _ ->
            nil
    end;
read_b64_img(_) -> nil.
